src/navigation_controller.ts

```diff
@@ -44,6 +45,16 @@ interface Scope {
   connection?: Connection;
 }
 
+/** Represents the current focus mode of the navigation controller. */
+enum NAVIGATION_FOCUS_MODE {
+  /** Indicates that no interactive elements of Blockly currently have focus. */
+  NONE = 'none',
+  /** Indicates either the toolbox or a flyout has focus. */
+  TOOLBOX_OR_FLYOUT = 'toolbox_or_flyout',
+  /** Indicates that the main workspace currently has focus. */
+  WORKSPACE = 'workspace',
```

Comment:
Handy since there's a chance we'll need to add `modal` or `widgetdiv` or similar as options.